tests/Unit/Utilities/Test_Functional.cpp

```diff
@@ -21,9 +21,28 @@
 #include "Utilities/TypeTraits.hpp"
 #include "tests/Utilities/MakeWithRandomValues.hpp"
 
+// IWYU pragma: no_include <algorithm>
+
 namespace funcl {
 using std::imag;
 using std::real;
+using std::max;
+using std::min;
+// Need to provide overloads of min/max for double with float because of the
+// tensor product of typelists in the tests.
```

Comment:
I'm afraid the structure of these tests doesn't actually create a `SPECTRE_TEST_CASE` when `MAKE_..._TEST` is called. These just create the`TestFuncEval...` structs. To make the test get called, you'll need to add the struct to the `std::make_tuple` line for `just_floating_binaries`. Sorry, this one is probably my fault for making test cases for which it is difficult to tell whether or not you actually manage to test something -- if you want this to be cleaned up, this can be an issue for me.

I'm also a bit confused that these are necessary. We have tests for `atan2` that just forward to the standard, and as far as I can tell the result is a quiet implicit cast. Do these not work because min and max are templates?

Reply:
Huh, I could've sworn I had those at some point. I'll add them, thanks :)

Yes, it tries to deduce the template parameter `T` but since it deduces it as `double` for one and `float` for the other it doesn't know which one to take.

I ended up just doing a loop over the single types passing random values of the same type as both arguments. Hopefully this is correct now :)